[Update Account Settings] Preferences, About, Security

kosmydel opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Update all pages in AccountSettings that use illustrations according to the new designs:
- Preferences,
- Security,
- About.

It also [this minor regression](https://github.com/Expensify/App/pull/35564#issuecomment-1933800492).

And it also adds missing icons in the Account Settings.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/35606
$ https://github.com/Expensify/App/issues/35607
$ https://github.com/Expensify/App/issues/35608
$ https://github.com/Expensify/App/issues/35905
$ https://github.com/Expensify/App/issues/36246
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console
- [ ] Go to Account Settings and verify.
  - [ ] Preferences Page
  - [ ] Security Page
  - [ ] About Page
  - [ ] Link icons on the LHN menu (Help & Go to Expensify Classic).
- [ ] Verify the new designs.
- [ ] Verify the translations (Spanish and English).

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

N/A

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console
- [ ] Tests from above

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
<img width="357" alt="Screenshot 2024-02-08 at 13 31 03" src="https://github.com/Expensify/App/assets/104823336/2baf9d5d-6d68-4129-894a-33a2e308b662">

<img width="354" alt="Screenshot 2024-02-08 at 11 09 55" src="https://github.com/Expensify/App/assets/104823336/4e7a94c9-a55a-448d-af3d-38a420911cb1">
<img width="349" alt="Screenshot 2024-02-08 at 13 31 13" src="https://github.com/Expensify/App/assets/104823336/594e99d1-d7f0-42de-8511-1a388a92625f">

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<img width="356" alt="Screenshot 2024-02-08 at 11 02 04" src="https://github.com/Expensify/App/assets/104823336/97fc26e4-c244-4da7-a926-d482ee172a5c">
<img width="348" alt="Screenshot 2024-02-08 at 13 32 48" src="https://github.com/Expensify/App/assets/104823336/e14403c7-9dea-46b9-a4fd-742dba5ad6fc">
<img width="349" alt="Screenshot 2024-02-08 at 13 32 57" src="https://github.com/Expensify/App/assets/104823336/fa64d47a-f13b-4bd8-bf7d-468e2199c79e">

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

![simulator_screenshot_7E45C5B7-1B1C-4A86-81FE-6F3D4AFEEDB5](https://github.com/Expensify/App/assets/104823336/35be1d4d-b87d-4997-b473-6b8b23154c22)

<!-- add screenshots or videos here -->
![simulator_screenshot_ED8AD3D3-CE0A-4B39-A511-FC2138B414B9](https://github.com/Expensify/App/assets/104823336/f6782f26-7f5a-495b-b07e-9bd761b0ea81)

![simulator_screenshot_10E8A427-D5F1-457C-BA27-1873D8CBE89A](https://github.com/Expensify/App/assets/104823336/d4f21ca4-ac0b-4f8a-b04a-fb2b09e7208f)

</details>

<details>
<summary>iOS: mWeb Safari</summary>

![simulator_screenshot_81C64E26-FDA9-4D44-978E-5E3AFF4C6F6A](https://github.com/Expensify/App/assets/104823336/aca57a3f-0318-4db8-addd-71ac64978267)

![simulator_screenshot_859F248E-65E6-4312-9E0C-9A21701E98C0](https://github.com/Expensify/App/assets/104823336/c74991e3-affc-4db9-babf-594c04bcc37d)

![simulator_screenshot_67B99887-3A09-41AC-A015-DF20ECF5D3A9](https://github.com/Expensify/App/assets/104823336/413dbff1-b5dd-4696-9ada-eff7e09383e2)

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/104823336/bcc08516-ade7-4495-8d8c-ca5855df513f


</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/104823336/586aed57-ca9f-4d0b-bf57-81197c4c9111


</details>

kosmydel committed:
redesign preferences page

kosmydel committed:
Merge branch 'Expensify:main' into update-account-settings/preferences

kosmydel committed:
new about page design

kosmydel commented on src/languages/en.ts:
```diff
@@ -1037,6 +1045,14 @@ export default {
         defaultPaymentMethod: 'Default',
     },
     preferencesPage: {
+        appSection: {
+            title: 'App preferences',
+            subtitle: 'That is when you can experience true joy, when you have no fear.',
+        },
+        testSection: {
+            title: 'Test preferences',
+            subtitle: 'That is when you can experience true joy, when you have no fear.',
+        },
```
We need a correct copy here and Spanish translations.

cc @mountiny, @shawnborton

shawnborton replied:
@mountiny @trjExpensify @Expensify/design how do you want to handle these? I think we need marketing to provide copy for each of the sections, and then we get the translation from there. Perhaps we should make a separate issue for that, or use the main tracking issue for this account settings project?

dannymcclain replied:
> I think we need marketing to provide copy for each of the sections, and then we get the translation from there.

Totally agree.

trjExpensify replied:
Yeah, agreed. Let's use the main issue. Added some suggestions and the `waiting for copy` label: https://github.com/Expensify/App/issues/35606#issuecomment-1927704175

trjExpensify replied:
@kosmydel you probably saw, but we got the copy updated in the OP mock. 👍

kosmydel replied:
Thanks! Do we have somewhere Spanish translation of it?

trjExpensify replied:
Here for you: https://github.com/Expensify/App/issues/35606#issuecomment-1930097774

kosmydel committed:
add english copy

kosmydel committed:
add Spanish copy

kosmydel committed:
Merge branch 'main' into update-account-settings/preferences

kosmydel committed:
rename

kosmydel committed:
small fixes

kosmydel committed:
fix icon

kosmydel committed:
fix illustration height

kosmydel committed:
Merge branch 'main' into update-account-settings/preferences

kosmydel committed:
fix statusbar color on mWeb iOS

kosmydel committed:
fix lint

blazejkustra commented on src/components/LottieAnimations/index.tsx:
```diff
@@ -1,7 +1,9 @@
+import type {AnimationKey} from './types';
 
-const DotLottieAnimations: Record<string, DotLottieAnimation> = {
+const DotLottieAnimations: Record<AnimationKey, DotLottieAnimation> = {
```
Let's try this, instead. This way you don't nee dot define `AnimationKey`
```ts
const DotLottieAnimations = {
    ExpensifyLounge: {
        file: require('@assets/animations/ExpensifyLounge.lottie'),
        w: 1920,
@@ -26,6 +28,7 @@ const DotLottieAnimations: Record<string, DotLottieAnimation> = {
        file: require('@assets/animations/PreferencesDJ.lottie'),
        w: 375,
        h: 240,
        backgroundColor: colors.blue500,
    },
...
    Coin: {
        file: require('@assets/animations/Coin.lottie'),
        w: 375,
        h: 240,
    },
} satisfies Record<string, DotLottieAnimation>;

blazejkustra commented on src/pages/settings/Preferences/PreferencesPage.js:
```diff
@@ -73,7 +72,6 @@ function PreferencesPage(props) {
                         isCentralPane
                         subtitleMuted
                         illustration={LottieAnimations.PreferencesDJ}
-                        illustrationBackgroundColor={theme.PAGE_THEMES[SCREENS.SETTINGS.PREFERENCES.ROOT].backgroundColor}
```
Remove const theme = useTheme(); in this file

kosmydel committed:
improve types

kosmydel committed:
fix workspace list empty state

kosmydel committed:
fix

kosmydel committed:
Merge remote-tracking branch 'origin-swm/update-account-settings/about' into update-account-settings/preferences

kosmydel committed:
fix About page status bar

filip-solecki committed:
Security page update

kosmydel committed:
fix lint issues

filip-solecki committed:
Adjust illustration background color logic

kosmydel committed:
add missing icons

kosmydel commented:
Hey @shawnborton, @trjExpensify, could you have a look and verify designs?

shawnborton commented:
cc @Expensify/design too!

mountiny commented:
Making a build for easier testing https://github.com/Expensify/App/actions/runs/7832379365

hayata-suenaga commented:
> The StatusBar is not updating after the theme is changed.

Let's fix this issue in this PR 👍 

> I can see that there are some other issues related to StatusBar

The other issues you mentioned in the other issue are already handled somewhere else. The only issue we're left with is the issue with updating the color when the theme changes.

mountiny commented:
I have asked for help about the status bar

dubielzyk-expensify commented:
I could be looking at the wrong thing in Figma, but I thought the spacing between the card and the app header was meant to be 0. It looks like in the mock above there's around 8-12px or so?
![CleanShot 2024-02-09 at 08 59 35@2x](https://github.com/Expensify/App/assets/145416827/e608c245-6633-426a-ad54-1c4fb27b6fc5)

What I'm seeing in Figma:
![CleanShot 2024-02-09 at 09 01 26@2x](https://github.com/Expensify/App/assets/145416827/a5327eaf-a77f-422f-ae2f-0d1b4644adcb)

kosmydel commented:
Thanks @ishpaul777! I've applied the changes and it works!

> I could be looking at the wrong thing in Figma, but I thought the spacing between the card and the app header was meant

Great catch @dubielzyk-expensify! I've removed this padding on all three screens, and on the Workspace List Page empty state.

<img width="1235" alt="Screenshot 2024-02-09 at 10 15 24" src="https://github.com/Expensify/App/assets/104823336/275f4340-5632-49df-bd12-305a828c9630">

kosmydel committed:
address review

kosmydel committed:
Merge branch 'Expensify:main' into update-account-settings/preferences

kosmydel committed:
add comment

kosmydel commented on src/components/CustomStatusBarAndBackground/index.tsx:
```diff
@@ -124,7 +124,8 @@ function CustomStatusBarAndBackground({isNested = false}: CustomStatusBarAndBack
         updateGlobalBackgroundColor(theme);
-    }, [isDisabled, theme]);
+        updateStatusBarStyle();
+    }, [isDisabled, theme, updateStatusBarStyle]);
```
[Done](https://github.com/Expensify/App/pull/35672/commits/782d2ca07d5090f44d8e052893e9c093b77613db)

kosmydel committed:
Merge branch 'Expensify:main' into update-account-settings/preferences

kosmydel commented:
@s77rt it is ready for review.

s77rt commented on src/pages/settings/Security/SecuritySettingsPage.tsx:
```diff
@@ -43,29 +42,43 @@ function SecuritySettingsPage() {
+            <HeaderWithBackButton
+                title={translate('initialSettingsPage.security')}
+                shouldShowBackButton={isSmallScreenWidth}
+                onBackButtonPress={() => Navigation.goBack()}
```
```suggestion
                onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS)}
```

kosmydel replied:
It was changed on main as well:
https://github.com/Expensify/App/commit/03896dc5897d256d8842e0f1d0bd692c70944a05

marcochavezf commented:
Added https://github.com/Expensify/App/issues/36246 to the list of fixed issues. @kosmydel can you merge from main? I've seen the perf-test check has been solved in other PRs by updating the PR from main

s77rt commented on src/styles/index.ts:
```diff
@@ -445,6 +445,9 @@ const styles = (theme: ThemeColors) =>
 
         textIvoryLight: {
```
```suggestion
        textVersion: {
```

textIvoryLight would be misleading as such names should refer to styles that are about colors only

kosmydel replied:
It was adressed, but I forgot to comment it.

kosmydel committed:
Merge branch 'Expensify:main' into update-account-settings/preferences

kosmydel commented:
@marcochavezf I have merged it, but I can't test it now as I'm AFK.

s77rt commented:
Version text is not centered 

<img width="572" alt="Screenshot 2024-02-09 at 4 19 24 PM" src="https://github.com/Expensify/App/assets/16493223/cd463e89-84ea-49de-87e4-e7762e577561">

s77rt commented:
WalletEmptyState still has top padding

<img width="572" alt="Screenshot 2024-02-09 at 4 32 07 PM" src="https://github.com/Expensify/App/assets/16493223/018a0cb8-3352-4d57-84c4-f62ad16fc202">

s77rt commented:
The status bar on mWeb still not fixed (https://github.com/Expensify/App/issues/35632)

<img width="572" alt="Screenshot 2024-02-09 at 5 22 35 PM" src="https://github.com/Expensify/App/assets/16493223/8272cae4-5333-4abf-a905-50cda3d696cd">

s77rt commented:
Refreshing preference page on mWeb does not have correct status bar color

https://github.com/Expensify/App/assets/16493223/88a5a4da-cd85-49c1-9b53-890805f2d38f

kosmydel commented on src/pages/settings/Preferences/PreferencesPage.js:
```diff
@@ -43,67 +43,93 @@ const defaultProps = {
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnablePickerAvoiding={false}
+            shouldEnableMaxHeight
```
Going to remove it.

kosmydel commented:
> WalletEmptyState still has top padding

It's out of scope of this issue.

> Refreshing preference page on mWeb does not have correct status bar color

It doesn't work on almost all pages.

kosmydel commented:
> The status bar on mWeb still not fixed

If I change the color here, the insets on Android break. It is being worked on [in other issues](https://github.com/Expensify/App/issues/35632#issuecomment-1934279437).

kosmydel commented on src/components/LottieAnimations/index.tsx:
```diff
@@ -26,6 +27,7 @@ const DotLottieAnimations: Record<string, DotLottieAnimation> = {
         file: require('@assets/animations/PreferencesDJ.lottie'),
         w: 375,
         h: 240,
+        backgroundColor: colors.blue500,
```
Should I move it anyway, and import `colors`? But personally I think, that background color is closely related to the illustration.

s77rt replied:
This is not.a blocker. Let's keep it as is for now

kosmydel replied:
Thanks! If internal decides we want to change it, I will do it as a follow-up!

mountiny replied:
I tmight be nice to add this as a follow up but I would say once we will need this dynamic which we dont right now as far as I know

kosmydel committed:
remove shouldEnableMaxHeight

kosmydel committed:
Merge branch 'main' into update-account-settings/preferences

shawnborton commented:
@dubielzyk-expensify @dannymcclain addressing the comment [above](https://github.com/Expensify/App/pull/35672#issuecomment-1935067886), which is similar to Danny's comment [here](https://github.com/Expensify/App/pull/36120#issuecomment-1936106406): 

I think I lean towards always having the 12px of padding/gap below the header. Thoughts?

kosmydel commented:
> @dubielzyk-expensify @dannymcclain addressing the comment [above](https://github.com/Expensify/App/pull/35672#issuecomment-1935067886), which is similar to Danny's comment [here](https://github.com/Expensify/App/pull/36120#issuecomment-1936106406):
> 
> I think I lean towards always having the 12px of padding/gap below the header. Thoughts?

Just keep in mind, that this issue solves deploy blocker. 

In a moment I and @filip-solecki are finishing up work for today and won't be able to help.

kosmydel committed:
Revert "address review"

This reverts commit 56fd6571a7540ddd3662df2fcd4338e4e3140cf5.

hayata-suenaga commented:
@kosmydel does your PR also fix [this issue](https://github.com/Expensify/App/issues/35632#issuecomment-1934279437)?

kosmydel commented:
> Hey @kosmydel, I am trying investigate the statusbar color issues quick question: can you please confirm after this PR i think the only page left that has different statusbar color from `appBg` is Sign in page right ?

In the Account Settings - yes. We are removing all illustrated headers. But not sure about some RHP pages / WS settings.

Currently, the home screen has also wrong status bar color, but changing it breaks the insets on android.

filip-solecki committed:
Rename text styles

kosmydel commented:
> @kosmydel does your PR also fix [this issue](https://github.com/Expensify/App/issues/35632#issuecomment-1934279437)?

No, it doesn't. Simply changing its configuration breaks android insets. More details and examples [here](https://github.com/Expensify/App/pull/35564#issuecomment-1926772435). @ishpaul777 apparently is working on it.

kosmydel commented:
The padding has been updated to `12px`. The Profile page padding is going to be changed in [this PR](https://github.com/Expensify/App/pull/36120). The Wallet page padding will be changed when re

s77rt commented:
@kosmydel Can you fix this https://github.com/Expensify/App/pull/35672#issuecomment-1936221262 today?

kosmydel commented:
> @kosmydel Can you fix this [#35672 (comment)](https://github.com/Expensify/App/pull/35672#issuecomment-1936221262) today?

I will have a quick look.

hayata-suenaga commented:
> > @ishpaul777 apparently is working on it.
> 
> not officially (haven't been assigned a issue yet) but i am just curious so investigating will put a proposal once i have something satisfactory

@ishpaul777 which issue are you taking about?

kosmydel committed:
fix centering

kosmydel commented:
> @kosmydel Can you fix this [#35672 (comment)](https://github.com/Expensify/App/pull/35672#issuecomment-1936221262) today?

I've fixed it. Can you verify it works?

s77rt commented:
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="489" alt="android" src="https://github.com/Expensify/App/assets/16493223/9172cad3-a5e2-4fb9-9f71-7681e9fccd29">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="489" alt="mweb-chrome" src="https://github.com/Expensify/App/assets/16493223/a884a72e-64af-473a-bea8-cac95cfb19e4">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
<img width="572" alt="ios" src="https://github.com/Expensify/App/assets/16493223/22a58689-64ee-4187-ba5c-4b4407ca76b0">

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
<img width="572" alt="mweb-safari" src="https://github.com/Expensify/App/assets/16493223/d20a875f-4f0e-4eb7-a14f-3856f666cdc5">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
<img width="1840" alt="web" src="https://github.com/Expensify/App/assets/16493223/353213c9-0101-4f54-a73b-d84869f0d8cd">

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
<img width="1312" alt="desktop" src="https://github.com/Expensify/App/assets/16493223/c13a4fe6-8bcd-4910-b006-8cdbd5f8fa34">

</details>

s77rt commented:
@kosmydel This https://github.com/Expensify/App/pull/35672#discussion_r1484527805 has not been addressed

kosmydel commented:
> @kosmydel This [#35672 (comment)](https://github.com/Expensify/App/pull/35672#discussion_r1484527805) has not been addressed

https://github.com/Expensify/App/pull/35672/commits/0375e276caadeca44e5d9bfc87629a4bc1ac65f0

kosmydel commented:
> @kosmydel This [#35672 (comment)](https://github.com/Expensify/App/pull/35672#discussion_r1484527805) has not been addressed

https://github.com/Expensify/App/pull/35672/commits/0375e276caadeca44e5d9bfc87629a4bc1ac65f0

filip-solecki committed:
Merge branch 'main' into update-account-settings/preferences

mountiny reviewed (approved):
Thanks for thorough testing, moving this ahead

mountiny commented on src/components/CustomStatusBarAndBackground/index.tsx:
```diff
@@ -116,15 +116,17 @@ function CustomStatusBarAndBackground({isNested = false}: CustomStatusBarAndBack
 
         updateGlobalBackgroundColor(theme);
-    }, [isDisabled, theme]);
+        updateStatusBarStyle();
```
The call to update status bar style is making the SignInPage test fail because the test does not mock the Navigation.isReady method. @filip-solecki @kosmydel we need to update the tests to support this.

mountiny reviewed (changes requested):
SignInPage tests need to be updated to mock the Navigation correctly  after the `updateStatusBarStyle` change

kosmydel committed:
Merge branch 'Expensify:main' into update-account-settings/preferences

kosmydel committed:
attempt to fix the SignInPage-perf test

kosmydel committed:
fix perf tests

kosmydel commented:
The performance tests are fixed. cc @mountiny

mountiny reviewed (dismissed):
Thank you

mountiny commented:
Sorry for not getting to this sooner, please ping me in slack once you resolve conflicts

filip-solecki committed:
Merge branch 'main' into update-account-settings/preferences

filip-solecki commented:
@mountiny conflicts resolved

kosmydel commented:
> ### Changes Suggested to resolve the Statusbar background color issue (not adding proposal because new changes needed are straightforward i belive can be done in the PR)

I believe that this is out of scope of this original issue. If we have dedicated PR (https://github.com/Expensify/App/pull/34409) for handling the StatusBar we should focus to fix all related issues there.

mountiny reviewed (approved):
That looks good thank you

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

filip-solecki commented on src/styles/theme/themes/light.ts:
```diff
@@ -105,7 +105,7 @@ const lightTheme = {
         [SCREENS.SETTINGS.PREFERENCES.ROOT]: {
-            backgroundColor: colors.blue500,
+            backgroundColor: colors.productLight100,
             statusBarStyle: CONST.STATUS_BAR_STYLE.LIGHT_CONTENT,
```
Fixed [here](https://github.com/Expensify/App/pull/36452)

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7911416146) to production by https://github.com/thienlnam in version: 1.4.41-12 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅
